cloudmarker/stores/filestore.py

```diff
@@ -0,0 +1,80 @@
+"""Filesystem store plugin."""
+
+
+import json
+import os
+
+
+class FileStore:
+    """A plugin to store records on the filesystem."""
+
+    def __init__(self, path='/tmp/cloudmarker'):
```

Comment:
I think this is okay. I would like to keep the mock plugins and this somewhat useful `filestore` plugin as simple as possible, so that we can refer to these while creating actual production-grade plugins.

[FHS 2.3](http://www.pathname.com/fhs/pub/fhs-2.3.html) guarantees the presence of [`/tmp`](http://www.pathname.com/fhs/pub/fhs-2.3.html#TMPTEMPORARYFILES), so this assumption is good for Unix-like systems.

On Windows, where `/tmp` may be absent, one would have to create a`\tmp` folder under the current drive to run this project with the default base configuration. But I am not thinking about supporting Windows explicitly at this time because that would require a little bit of additional effort in terms of testing.

We can revisit this decision regarding`/tmp/cloudmarker` vs. `tempfile` and Windows support at a later time.

Reply:
**PS**: Didn't know there is a `tempfile` module too, and that too native

Adding on @s-nirali's comment 

MacOS python3 
```
>>> import tempfile
>>> tempfile.gettempdir()
'/var/folders/30/qpxs8kwj3jzc612r2gsq17zwc2yvq5/T'
```

Debain9 python3 
```
>>> import tempfile
>>> tempfile.gettempdir()
'/tmp'
```

Windows python3

![image](https://user-images.githubusercontent.com/5807820/50396945-137c2200-0793-11e9-8763-73c99ce9fa23.png)

The Linux one and Windows ones are still doable but Unix is just plain wierd.

![zyeco](https://user-images.githubusercontent.com/5807820/50397299-77074f00-0795-11e9-9b05-10bd6fa6b1d4.jpg)

Reply:
@susam was the one who introduced me to the `tempfile` module in one of his reviews for my code. :smile:

Reply:
Here is another relevant comic about Python.

[![XKCD #353 on Python](https://imgs.xkcd.com/comics/python.png)](https://xkcd.com/353/)

> Title text: I wrote 20 short programs in Python yesterday.  It was wonderful.  Perl, I'm leaving you.

[Source: <https://xkcd.com/353/>]

Back in 2007 when this comic was published, this was very relevant because Perl was still more popular and Python was about to overtake it in terms of popularity. Python was steadily gaining fans as well as Perl-to-Python converts due to how simple programming felt while writing Python code due to its clean syntax and extensive standard library.

![PYPL graph of Python vs. Perl popularity](https://i.imgur.com/de3wFWB.png)

[Source: <http://pypl.github.io/PYPL.html>]

Now Python has become so popular that we can simply `pip install` and `import` machine learning libraries. The Python ecosystem has come a long way in the last ten years.